uint8 red
uint8 green
uint8 blue
float64 x_location
float64 y_location
float64 rough_size